playwright/src/test/java/com/microsoft/playwright/TestPageClock.java

```diff
@@ -310,6 +311,7 @@ void shouldNotRunTimeBeforePopupOnPause(Page page, Server server) {
     Page popup = page.waitForPopup(() -> {
       page.evaluate("url => window.open(url)", server.PREFIX + "/popup.html");
     });
+    popup.waitForURL(server.PREFIX + "/popup.html");
```

Comment:
Looks like upstream does waitForLoadState? https://github.com/microsoft/playwright/blob/77dee44984af1cd5b7c1ec4f6ed6fe022c7c55e6/tests/library/page-clock.spec.ts#L357

Reply:
Wait for url does wait for load state too by default and also ensures we are not on about:blank (unlikely).